x/pairing/types/relay_finalization.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tendermintcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+type RelayFinalization struct {
+	Exchange RelayExchange
+	Addr     sdk.AccAddress
+}
+
+func NewRelayFinalization(exch RelayExchange, addr sdk.AccAddress) RelayFinalization {
+	return RelayFinalization{Exchange: exch, Addr: addr}
+}
+
+func (rf RelayFinalization) GetSignature() []byte {
+	return rf.Exchange.GetSignature()
+}
+
+func (rf RelayFinalization) PrepareForSignature() []byte {
+	relaySessionHash := tendermintcrypto.Sha256(rf.Exchange.Request.RelaySession.CalculateHashForFinalization())
+	latestBlockBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(latestBlockBytes, uint64(rf.Exchange.Reply.LatestBlock))
```

Comment:
these 2 lines (26-27) repeat several times - please add a helper and replace with e.g. `blockBytes := sigs.EncodeBlock(rf.Exchange.Replay.LatestBlock)` (and elsewhere too)

Reply:
done. see https://github.com/lavanet/lava/commit/1bee279f00a0434b68b241f05cfb240c05663c1c

Reply:
nice!